core/process.dm

```diff
@@ -197,19 +199,15 @@ datum/controller/process/proc/scheck(var/tickId = 0)
 
 	// For each tick the process defers, it increments the cpu_defer_count so we don't
 	// defer indefinitely
-	if (main.getCurrentTickElapsedTime() > main.timeAllowance)
-		sleep(world.tick_lag*1)
+	if (world.tick_usage > 100 || main.world.tick_usage > tick_start + tick_allowance)
```

Comment:
more like if (world.tick_usage > 100 || world.tick_usage > tick_start + tick_allowance)

not sure how that snuck in there